docs/manual/build_system.qbk

```diff
@@ -6,44 +6,105 @@
 
 The buildsystem for __hpx__ is based on __cmake__. CMake is a cross-platform
 build-generator tool. CMake does not build the project, it generates the files
 needed by your build tool (GNU make, Visual Studio, etc) for building __hpx__.
 
-This section gives an introduction on how to use our build system to build HPX
-and how to use HPX in your own projects.
+This section gives an introduction on how to use our build system to build __hpx__
+and how to use __hpx__ in your own projects.
 
 [include build_system/cmake_basics.qbk]
 [include build_system/prerequisites.qbk]
 
-[section:building_hpx Building HPX]
+[section:building_hpx Building __hpx__]
+
+[heading:info Basic Information]
+
+Once __cmake__ has been run, the build process can be started. The __hpx__ build process
+is highly configurable through __cmake__ and various __cmake__ variables influence
+the build process. The build process consists of the following parts:
+
+* [*The __hpx__ core libraries]: This forms the basic set of __hpx__ libraries. The generated targets are:
+  * hpx: The core __hpx__ library (always enabled).
+  * hpx_init: The __hpx__ initialization library that applications need to link
+    against to define the __hpx__ entry points (disabled for static builds).
+  * iostreams_component: The component used for (distributed) IO (always enabled).
+  * component_storage_component: The component needed for migration to persistant storage.
+  * unordered_component: The component needed for a distributed hash table.
+* [*__hpx__ Examples]: This target is enabled by default and builds all __hpx__ examples
+  (disable by setting HPX_WITH_BUILD_EXAMPLES=Off). __hpx__ examples are part of the
+  'all' target and are included in the installation if enabled.
+* [*__hpx__ Tests]: This target builds the __hpx__ test suite and is enabled by default
+  (disable by setting HPX_WITH_BUILD_TESTS=Off). They are not built by the 'all'
+  target and have to be built separately.
+* [*__hpx__ Documentation]: This target builds the Documentation, this is not
+  enabled by default. For more information see
+  [link hpx.manual.build_system.building_hpx.doc_tools Setting up the __hpx__ Documentation Tool Chain].
+
+For a complete list of available __cmake__ variables that influence the build of
+__hpx__ see [link hpx.manual.build_system.building_hpx.cmake_variables CMake Variables used to configure __hpx__].
+
+The variables can be used to refine the recipes that can be found
+[link hpx.manual.build_system.building_hpx.build_recipes here] which show some basic steps on
+how to build HPX for a specific platform
+
+In order to use __hpx__, only the core libraries are required (the ones marked
+as optional above are truly optional). When building against __hpx__, the __cmake__
+variable `HPX_LIBRARIES` will contain hpx and hpx_init (for pkgconfig, those are
+added to the `Libs` sections). In order to use the optional libraries, you need
+to specify them as link dependencies in your build.
+
```

Comment:
`HPX_LIBRARIES` will be available only after invoking `find_package(HPX)` (and possibly other cmake macros). It might be useful to have a minimal example for an external cmake-file here (or a link to the example) for illustration purposes.